pkg/ast/pipesearch/wsSearchHandler.go

```diff
@@ -366,6 +377,28 @@ func processMaxScrollComplete(conn *websocket.Conn, qid uint64) {
 	}
 }
 
+func UpdateWSResp(wsResponse *structs.PipeSearchWSUpdateResponse, qType structs.QueryType, qid uint64) error {
+	getResp := query.GetPipeResp(qid)
+	if getResp == nil {
+		return fmt.Errorf("qid=%d, UpdateWSResp: failed to get new query pipeline response", qid)
+	}
+	switch qType {
+	case structs.SegmentStatsCmd, structs.GroupByCmd:
+		wsResponse.MeasureResults = getResp.MeasureResults
+		wsResponse.MeasureFunctions = getResp.MeasureFunctions
+		wsResponse.GroupByCols = getResp.GroupByCols
+		wsResponse.BucketCount = getResp.BucketCount
+		wsResponse.ColumnsOrder = getResp.ColumnsOrder
+		wsResponse.Qtype = qType.String()
+	case structs.RRCCmd:
+		wsResponse.Hits = getResp.Hits
+		wsResponse.AllPossibleColumns = getResp.AllPossibleColumns
+		wsResponse.Qtype = qType.String()
+		wsResponse.ColumnsOrder = getResp.ColumnsOrder
+	}
```

Comment:
You should have a `default` (and maybe a `case structs.GroupbyCmd`) here, as Lindy mentioned

Reply:
addressed in the latest commit.